validator-flow.md

```diff
@@ -22,18 +22,23 @@ On start of every epoch, validator should ask for attester duties for epoch + 1.
 Result are array of objects with validator, his committee and attestation slot.
 
 Attesting:
-1. [Upon receiving duty check if aggregator](https://github.com/ethereum/eth2.0-specs/blob/v0.11.3/specs/phase0/validator.md#aggregation-selection)
-    - Validator is aggregator
-    - [Ask beacon node to subscribe to your subnet](#/ValidatorRequiredApi/subscribeToBeaconCommitteeSubnet)                                             
-2. on start of attesting slot, [fetch AttestationData](#/ValidatorRequiredApi/produceAttestationData)
-3. wait for AttestationData block (TBD)
-    - max wait: SECONDS_PER_SLOT / 3 * 1000
-4. [submit Attestation](#/ValidatorRequiredApi/submitPoolAttestations) (AttestationData + aggregation bits)
+1. Upon receiving duty, have beacon node prepare committee subnet
+    - [Check if aggregator by computing `slot_signature`](https://github.com/ethereum/eth2.0-specs/blob/v0.12.2/specs/phase0/validator.md#attestation-aggregation)
+    - [Ask beacon node to prepare your subnet](#/ValidatorRequiredApi/prepareBeaconCommitteeSubnet)
+      -- Note, validator client only needs to submit one call to
+      `prepareBeaconCommitteeSubnet` per committee/slot it's validators have
+      been assigned to. If any validators are aggregators, be sure to use the aggregator's
+      `slot_signature` to properly signal aggregation to the beacon node
+2. Wait for new BeaconBlock for the assigned slot (either stream updates or poll)
+    - Max wait: `SECONDS_PER_SLOT / 3 * 1000` into the assigned slot
```

Comment:
Why the factor of 1000 here?

Reply:
not sure. was there before the changes. I think just keeping it in seconds makes sense